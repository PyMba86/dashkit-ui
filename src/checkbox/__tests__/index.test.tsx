import * as React from 'react';
import { mount } from 'enzyme';
import Checkbox from '..';
const prefixCls = 'dk-checkbox';

describe('Checkbox', () => {
  test('should react while click', () => {
    const checkbox = mount(<Checkbox defaultChecked>Option</Checkbox>);
    expect(checkbox.find(`.${prefixCls}`).hasClass(`${prefixCls}-checked`)).toBeTruthy();
    checkbox.find(`.${prefixCls}-input`).simulate('change', {
      target: {
        checked: false,
      },
    });
    checkbox.update();
    expect(checkbox.find(`.${prefixCls}`).hasClass(`${prefixCls}-checked`)).toBeFalsy();
  });
});
