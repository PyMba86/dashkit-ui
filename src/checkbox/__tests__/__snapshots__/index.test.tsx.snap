// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup correctly sets "checked" attribute 1`] = `
<div
  className="dk-checkbox-group"
>
  <label
    className="dk-checkbox dk-checkbox-group-item"
  >
    <input
      checked={false}
      className="dk-checkbox-input"
      onChange={[Function]}
      type="checkbox"
      value="1"
    />
    <i
      className="dk-checkbox-indicator"
    />
  </label>
  <label
    className="dk-checkbox dk-checkbox-group-item"
  >
    <input
      checked={false}
      className="dk-checkbox-input"
      onChange={[Function]}
      type="checkbox"
      value="3"
    />
    <i
      className="dk-checkbox-indicator"
    />
  </label>
</div>
`;

exports[`CheckboxGroup renders correctly when deeply nested 1`] = `
<div
  className="dk-checkbox-group"
>
  <div>
    <div>
      <label
        className="dk-checkbox dk-checkbox-group-item"
      >
        <input
          checked={false}
          className="dk-checkbox-input"
          onChange={[Function]}
          type="checkbox"
          value="1"
        />
        <i
          className="dk-checkbox-indicator"
        />
      </label>
      <label
        className="dk-checkbox dk-checkbox-group-item"
      >
        <input
          checked={false}
          className="dk-checkbox-input"
          onChange={[Function]}
          type="checkbox"
          value="3"
        />
        <i
          className="dk-checkbox-indicator"
        />
      </label>
    </div>
  </div>
</div>
`;
